### PR TITLE
Adding voice behaviour

### PR DESCRIPTION
**Pull request by ERAGON007**
I made some changes in the repo, but not completed yet they need to be reviewed

### CODE REVIEW

File: SlipeServer.Packets/Definitions/Voice/VoiceDataPacket.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            this.Buffer = sourceBuffer;
+            
+            this.SourceElementId = elementId;
+        }
+        
+        public override byte[] Write()
+        {
+            var builder = new PacketBuilder();
+
+            // Write the source player id
```
Comment (NanoBob) [inline SlipeServer.Packets/Definitions/Voice/VoiceDataPacket.cs:31]: These comments can be removed, the code is self explanatory enough

File: SlipeServer.Server/Behaviour/VoiceBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            this.server = server;
+            this.logger = logger;
+
+            server.PlayerJoined += (player) =>
+            {
+                player.OnVoiceData += (sender, args) =>
+                {
+                    var packet = new VoiceDataPacket(player.Id, args.DataBuffer);
+                    server.BroadcastPacket(packet);
```
Comment (NanoBob) [inline SlipeServer.Server/Behaviour/VoiceBehaviour.cs:26]: Broadcasting also sends the packet to the client it originated from, is that desired?

File: SlipeServer.Server/Configuration.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public class Configuration
 
         public float ExplosionSyncDistance { get; set; } = 400;
 
+        public bool IsVoiceEnabled { get; set; } = true;
```
Comment (NanoBob) [inline SlipeServer.Server/Configuration.cs:36]: Personally I think voice should default to being disabled

File: SlipeServer.Server/Elements/Player.cs [hunk 4]
```diff
@@ -46,6 +46,7 @@ public byte WantedLevel
         public bool IsOnFire { get; set; }
         public bool IsSyncingVelocity { get; set; }
         public bool IsStealthAiming { get; set; }
+        public bool IsMuted { get; set; }
```
Comment (NanoBob) [inline SlipeServer.Server/Elements/Player.cs:49]: Players can also be muted for chat purposes, so I would rename this to:
`IsVoiceMuted` (and later add a `IsChatMuted`)

**Comment by ERAGON007** [comment]
Fixed conflicts with #138 and done implementing of voice support (needs further testing with 2 clients connected) before merge

File: SlipeServer.Server/Elements/Player.cs [hunk 5]
```diff
@@ -118,6 +120,16 @@ public void Kill(WeaponType damageType = WeaponType.WEAPONTYPE_UNARMED, BodyPart
             this.Kill(null, damageType, bodyPart);
         }
 
+        public void VoiceDataStart(byte[] voiceData)
+        {
+            this.OnVoiceData?.Invoke(this, new PlayerVoiceStartArgs(this, voiceData));
```
Comment (NanoBob) [inline SlipeServer.Server/Elements/Player.cs:125]: The only suggestion I would still add for this PR is using `IsVoiceMuted`, and only triggering the `OnVoiceData` event if the user is not muted.

**Review by NanoBob — CHANGES_REQUESTED** [review]

**Review by NanoBob — APPROVED** [review]

**Pull request merged**